src/rxToStream.ts

```diff
@@ -12,59 +11,70 @@ export function rxToStream<T extends Streamable>(
     options: stream.ReadableOptions = { encoding: 'utf8' },
     onError?: (error: Error, readable: stream.Readable) => void
 ): stream.Readable {
+    return new ReadableObservableStream(options, src, onError);
+}
+
+class ReadableObservableStream<T> extends stream.Readable {
+    constructor(
+        options: stream.ReadableOptions,
+        private _source: Observable<T>,
+        private _onError: ((error: Error, readable: stream.Readable) => void) | undefined,
+    ) {
+        super(options);
+    }
 
-    const trigger = new Subject<void>();
-    let depth = 0;
-    const maxDepth = 100;
+    private _isOpen = false;
+    private _hasError = false;
+    private _error: any;
+    private _waiting = 0;
+    private _subscription: Subscription;
+    private _buffer: T[] = [];
 
-    const readable = new stream.Readable({
-        ...options,
-        read: () => {
-            trigger.next();
-        },
-    });
+    _read() {
+        const { _buffer } = this;
 
-    function close(err?: Error) {
-        try {
-            if (err && onError)
-                onError(err, readable);
-        } finally {
-            interval(1).pipe(take(1)).subscribe(
-                () => readable.push(null)
-            );
+        if (!this._subscription) {
+            this._subscription = this._source.subscribe({
```

Comment:
What do you expect to happen for Observables that are hot or become hot before the first call to _read()? It seems like the data will get lost.